drivers/sht2x/sht2x.c

```diff
@@ -0,0 +1,259 @@
+    if (result < 1 || (user_reg & ~SHT2X_USER_RESERVED_MASK) != 0x02) {
+        i2c_release(BUS);
+        DEBUG("[sht2x] init - error: can't read user_reg or wrong value [0x%02x]\n", (int)user_reg);
+        return SHT2X_NODEV;
+    }
+
+    // write desired resolution to user register
+    // while keeping the rest of the register the same
+    // (only if current resolution != desired resolution)
+    DEBUG("[sht2x] init: setting resolution\n");
+    if ((user_reg & SHT2X_USER_RESOLUTION_MASK) != dev->params.resolution) {
+        result = i2c_write_reg(BUS, ADDR, SHT2X_USER_REG_W,
+            (dev->params.resolution | (user_reg & ~SHT2X_USER_RESOLUTION_MASK)));
+        if (result < 1) {
+            i2c_release(BUS);
+            DEBUG("[sht2x] init - error: user register write failed\n");
+            return SHT2X_NODEV;
+        }
+    }
+
+    DEBUG("[sht2x] init: releasing I2C\n");
+    i2c_release(BUS);
+
+    DEBUG("[sht2x] init: successful\n");
+
+    return SHT2X_OK;
+}
+
+// Returns °C (x100)
+int16_t sht2x_read_temp(sht2x_t *dev)
+{
+    uint8_t result;
+    char buffer[3];
+    
+    uint16_t st;
+    float coeff;
+    float temp;
+
+    assert(dev);
+
+    // acquire I2C bus
+    DEBUG("[sht2x] read_temp: acquiring I2C\n");
+    i2c_acquire(BUS);
+
+    // trigger measurement
+    DEBUG("[sht2x] read_temp: triggering no hold master temp read\n");
+    result = i2c_write_byte(BUS, ADDR, SHT2X_TRIG_T_MEASUREMENT_POLL);
+    if (result < 1) {
+        i2c_release(BUS);
+        DEBUG("[sht2x] read_temp - error: i2c write failed\n");
+        return SHT2X_NODEV;
+    }
+
+    // sleep for the typical time it takes to complete the measurement
+    // this depends on the resolution and is taken from the datasheet
+    switch(dev->params.resolution) {
+        case SHT2x_RES_12_14BIT:
+            xtimer_usleep(66 * 1000);
+            break;
+        case SHT2x_RES_8_12BIT:
+            xtimer_usleep(17 * 1000);
+            break;
+        case SHT2x_RES_10_13BIT:
+            xtimer_usleep(33 * 1000);
+            break;
+        case SHT2x_RES_11_11BIT:
+            xtimer_usleep(9 * 1000);
+            break;
+        default:
+            // this case is impossible
+            i2c_release(BUS);
+            DEBUG("[sht2x] read_temp - error: you what???\n");
+            return SHT2X_OTHERERROR;
+    }
+    
+    uint8_t retries = 0;
+    result = i2c_read_bytes(BUS, ADDR, buffer, 3);
+    while (result < 3) {
+        xtimer_usleep(1000);
+        retries++;
+        if (retries > MAX_RETRIES) {
+            i2c_release(BUS);
+            DEBUG("[sht2x] read_temp - error: sensor read timed out\n");
+            return SHT2X_READERROR;
+        }
+        result = i2c_read_bytes(BUS, ADDR, buffer, 3);
+    }
+
+    DEBUG("[sht2x] read_temp: releasing I2C\n");
+    i2c_release(BUS);
+
+    // ** TODO **
+    // verify checksum
+
+    st = buffer[0] << 8 | buffer[1];
+    coeff = st / 65536.0;
+
+    temp = -46.85 + 175.72*coeff;
```

Comment:
for instance here, you can easily scale by factor `* 100` to get rid of floating points in `-46.85` and `175.72`